File: plotMapAllocation_stage1.py
import matplotlib.pyplot as plt


def plotMapAllocation_stage1(X, n_rounds, na, color, name):
    for i in range(na):
        xx = X[:, i, :]
        xx = xx[:, xx[0, :] != 0]

        if i == 0:
            plt.plot(xx[0, :], xx[1, :], ':', color=color[i], linewidth=2, label=name)
        else:
            plt.plot(xx[0, :], xx[1, :], ':', color=color[i], linewidth=2)
    # plt.legend()

File: original_matlab_code/plotMapAllocation_stage1.m
function plotMapAllocation_stage1(X, n_rounds, na, color, name)

    for i = 1:na      
%         xx = reshape(X(:,i,:),[4,n_rounds+1]);
        xx = reshape(X(:,i,:),[4,size(X,3)]);
        xx(:,find(xx(1,:)==0))=[];
        plot(xx(1,:),xx(2,:), ':', 'Color', color(i,:), 'LineWidth', 2, 'DisplayName', name);
    end

end